### PR TITLE
Add the release notes style guide

### LINKED ISSUE #2111: [5pt] Describe English style to use in release notes
**Epic: [Release notes](https://github.com/tarantool/doc/issues/2570)**

# Describe English style to use in release notes

**Product:** Tarantool (mostly about tarantool, but I guess it is applicable for all products)
**Since:** N/A
**Audience/target:** Tarantool developers and documentarians
**Root document:** Somewhere around our documentation guides
**SME:** @ NickVolynkin, @ Gerold103, @ patiencedaur

# Details

See also: #1285

The discussion was started [here](https://github.com/tarantool/tarantool/pull/5935#discussion_r605901476). I'll cite:

> @Gerold103:
>
> You might notice that we use imperative in all changelog records. Would be good to be consistent.

> @Totktonada:
>
> In the past release notes:
>
> - Imperative mood is used when we start to do X: say, 'Deploy packages for Fedora 32 (gh-4966)'.
> - Past tense is used to describe a past (changed / fixed) behaviour: say, 'Fixed a false positive panic when yielding in debug hook (gh-5649)'.
> - Present tense is used to describe a new behaviour: say, 'Data changes in read-only mode are now forbidden (gh-5231)'.
>
> Unlike using an imperative mood, this approach results in a text that is 'written for a human', like an article about changes. For me, it looks good.
>
> However, I'm not an expert neither in grammar, nor in documentation writting, so it is better to ask @tarantool/doc team.
>
> > You might notice that we use imperative in all changelog records. Would be good to be consistent.
>
> There is no consistency in current unreleased changelog entries, but I hope we'll proof-read them before a next release and adjust to a certain style. We can elaborate some preliminary agreements with the documentation team here.

To be honest, I really don't spin around grammatic rules enough to insist on my vision. However it looks meaningful for me and if we'll agree on another set of rules, I would like to know a reasoning behind.

# Definition of done

- [ ] There is a page with recommendations how to write grammatically correct release notes.
- [ ] @Totktonada understands how to write changelog entries.

# Tentative directions

- [ ] Make a list of the most common problems documentarians *and translators* face with RNs. See #1285
- [ ] Analyze the corresponding sections (if they exist) in Microsoft and Google style guides
- [ ] Read the article [provided as a guideline](https://uxdesign.cc/the-art-of-writing-great-release-notes-6607e22efae1) in #1285
- [x] Outline the contents
- [ ] Start writing :)

---

]Notes from @ lenkis:

Problems with OS Tarantool release notes (RN):

1. No style guide for RN, so writers need to brush up the draft a lot (e.g. RN items are phrased differently).
TODO: Introduce a simple style guide for all RN contributors.

2. No automatic integration of RN into the manual, so writers need to do it manually.
TODO: If we continue mirroring RN in the manual, automate md-to-rst import.

3. Too wordy RN section in the manual, which adds no value beyond RN text at GH.
TODO: Think of making the most of RN as a community communication tool. Here is an article for inspiration: https://uxdesign.cc/the-art-of-writing-great-release-notes-6607e22efae1

---

Draft Notion document with ideas: https://www.notion.so/tarantool/WIP-Release-notes-workflow-b4292dbf8fbd41d58a1983546de4a879

### PR DESCRIPTION
**Pull request by patiencedaur**
Closes #2111 

Formulate some language rules. Ideally, they would apply to any Tarantool product release.

@Totktonada: is this a good place for such guidelines? This is more of an internal document after all—the audience are our developers.

### CODE REVIEW

**Comment by Totktonada** [comment]
NB: The issue is #2111.

File: doc/contributing/release_notes.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+==========================
+
+Release notes are generated by a script from individual changelog strings.
+Below are some best practices to make these changelog strings consistent, neat, and human-oriented.
+
+*   Use the past tense to describe changed or fixed behavior.
+    Example: 'Fixed a false positive panic when yielding in debug hook (gh-5649)'.
+*   Use the present tense to describe new behavior.
+    Example: 'Data changes in read-only mode are now forbidden (gh-5231)'.
+*   Start with a capital letter, end with a period.
```
Comment (Totktonada) [inline doc/contributing/release_notes.rst:11]: Cited from the issue:

> Imperative mood is used when we start to do X: say, 'Deploy packages for Fedora 32 (gh-4966)'.

Is it forbidden? If so, how to write this change description? If not, please, include.

Comment (patiencedaur) [reply doc/contributing/release_notes.rst:11]: I'm not sure I understand why you need the imperative mood in the release notes. This document is not telling the users what to do but describing what has been done.
'Deploy packages for Fedora 32 (gh-4966)'—doesn't it mean that "packages have been deployed"?

File: doc/contributing/release_notes.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+How to write release notes
+==========================
+
+Release notes are generated by a script from individual changelog strings.
```
Comment (Totktonada) [inline doc/contributing/release_notes.rst:4]: It is applicable only for tarantool/tarantool repository, but the article is about style and I guess it should be general enough to apply for any project. I would not mention any technical details: generation scripts, file location, file formatting, etc; and would concentrate on the style here.

File: doc/contributing/release_notes.rst [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+Below are some best practices to make these changelog strings consistent, neat, and human-oriented.
+
+*   Use the past tense to describe changed or fixed behavior.
+    Example: 'Fixed a false positive panic when yielding in debug hook (gh-5649)'.
+*   Use the present tense to describe new behavior.
+    Example: 'Data changes in read-only mode are now forbidden (gh-5231)'.
+*   Start with a capital letter, end with a period. 
+
+Note that these guidelines differ from commit message best practices that suggest using the imperative mood.
```
Comment (Totktonada) [inline doc/contributing/release_notes.rst:13]: Technically the imperative mood is suggested for commit message *titles*.

File: doc/contributing/release_notes.rst (hunk 1 above)
Comment (Totktonada) [reply doc/contributing/release_notes.rst:11]: It means that a necessary machinery is integrated into the repository, so the release will be deployed for given OS, when it'll be released. (Also it implicitly means that we start to test commits against this OS to don't break anything here.) The outcome for a user: download instructions on the website will work for given OS after the release. If you don't like this wording, can you propose one that satisfies you? So I would able to compare and either agree or give arguments against.

Comment (patiencedaur) [reply doc/contributing/release_notes.rst:11]: My proposition (if I understood you right):
(Tarantool) packages will be available for Fedora 32 on release.

"Deploy" can be understood both as an imperative and as present tense, which is confusing.

Comment (Totktonada) [reply doc/contributing/release_notes.rst:10]: So it'll look so in the release notes:

> * Packages will be available for Fedora 32 on release (gh-4966).

Release notes are published at release, so packages should be already available. So 'will be' and 'on release' looks strange in this context for me.

If we're going this way, maybe someting like this?

> * Fedora 32 is now supported. Per-commit testing and deployment of prereleases and releases to the YUM repository is added (gh-4966).

Whether this variant looks clear for you?

----

Is seems, we should collect examples for such situations: where we start to discuss how to better reflect a change. I propose to add an example for a new supported distribution after this discussion.

File: doc/contributing/release_notes.rst (hunk 2 above)
Comment (Totktonada) [reply doc/contributing/release_notes.rst]: @patiencedaur Please, respond at comments, when they're resolved. This one looks resolved now.

File: doc/contributing/release_notes.rst (hunk 3 above)
Comment (Totktonada) [reply doc/contributing/release_notes.rst]: Resolved now.

**Comment by Totktonada** [comment]
> @Totktonada: is this a good place for such guidelines? This is more of an internal document after all—the audience are our developers.

I would not bother now about suitable/good/best place. Let's start here and decide later to move (or don't move).

BTW, the audience is broader: external contributors should follow the same guidelines.

File: doc/contributing/release_notes.rst (hunk 1 above)
Comment (patiencedaur) [reply doc/contributing/release_notes.rst:10]: Regarding my first example: I meant the Fedora release, not the Tarantool release.

Regarding the second example:

- I would start the second sentence with "Added" as per other rules.
- I still don't understand why we're explaining to the reader what we are currently doing, or started doing. Release notes are about what has already been done, finished, included in the release. Could you probably explain it in Russian and/or in an oral conversation?

Here is another attempt—does the following make sense?
"Added packages for the upcoming Fedora 32 release."

Comment (Totktonada) [reply doc/contributing/release_notes.rst:10]: We usually add packages, when a distribution is already released.

> I would start the second sentence with "Added" as per other rules.

Ok.

> I still don't understand why we're explaining to the reader what we are currently doing, or started doing.

We don't, as I see. Ok, let's discuss voicely.

The result:

> * Fedora 32 is supported now. Added per-commit testing and deployment of prereleases and releases to the YUM repository (gh-4966).

Comment (Totktonada) [reply doc/contributing/release_notes.rst:10]: After the voice discussion:

> * Fedora 32 is supported now. Added per-commit testing, updated YUM repositories (gh-4966).

Comment (Totktonada) [reply doc/contributing/release_notes.rst]: I ask you again to respond to comments, when they're fixed.

This one is fixed now, I'm closing the discussion.

**Review by Totktonada — APPROVED** [review]
I have no objections.

**Review by Gerold103 — APPROVED** [review]
